### PR TITLE
Group 3

### PR DESCRIPTION
**Pull request by eliloz**
**Team Members:**
- Lana
- Eli
- Corey

### CODE REVIEW

File: simple_database/main.py [hunk 1]
```diff
@@ -1,5 +1,219 @@
...
+        self.db.save()
+
+    #def query():
+    
+    
+    #def describe():
+
+
+    def all(self):
+        return [row for row in self.rows]
```
Comment (lalvarezguillen) [inline simple_database/main.py]: I wonder if that works for you. I think the tests are calling next on the return value of all(). And I don't think lists have a next.

``` python
#Maybe:
return iter([n for n in stuff])
#or a generator
return ( n for n in stuff)
```

File: simple_database/main.py [hunk 2]
```diff
@@ -1,5 +1,219 @@
...
+        for item in self.columns:
+            if isinstance(args[i], eval(item.get('type'))):
+                pass
+            else:
+                raise ValidationError('Invalid type of field "{}": Given "{}", expected "{}"'.format(item.get('name'), type(args[i]).__name__, item.get('type')))
+                
+            i += 1
+
+        # Create new row (list) in database file
+        self.rows.append([arg for arg in args])
```
Comment (lalvarezguillen) [inline simple_database/main.py:100]: If each row is represented by a list:
    1.- Querying is gonna be a little hacky
    2.- I think some tests will fail. There's a test querying a row property, author.id. So rows are kind of required to be objects

**Comment by lalvarezguillen** [comment]
Hey G3! I see Travis is playing mean with you. David and I are supposed to review your solution. I left a couple of observations in your code

File: simple_database/main.py [hunk 3]
```diff
@@ -1,5 +1,140 @@
...
+        try:
+            os.stat(BASE_DB_FILE_PATH)
+        except:
+            os.mkdir(BASE_DB_FILE_PATH) 
+
+        if action == 'connect' and os.path.isfile(self.file_path):
+            self.load()
+        if action == 'create' and os.path.isfile(self.file_path):
+            raise ValidationError('Database with name "{}" already exists.'.format(db_name))
+
```
Comment (clarkjdj) [inline simple_database/main.py:23]: I like this database interface for create and connect. Very clean.

File: simple_database/main.py [hunk 4]
```diff
@@ -1,5 +1,140 @@
...
+            for index, column in enumerate(self.columns):
+                if column['name'] == key:
+                    i = index
+            # Return row matching value at row index
+            for row in self.rows:
+                if row[i] == val:
+                    yield Row(row, self.columns)
+
+    def describe(self):
+        return self.columns
```
Comment (clarkjdj) [inline simple_database/main.py:119]: Is this going to return a list of names or a list of dictionaries?

**Pull request closed without merging**